C/tests/c4DatabaseTest.cc

```diff
@@ -1011,6 +1014,69 @@ static void testOpeningOlderDBFixture(const string& dbPath, C4DatabaseFlags with
     CHECK(c4db_delete(db, WITH_ERROR()));
 }
 
+N_WAY_TEST_CASE_METHOD(C4DatabaseTest, "Database Enumerator with Conflicted Option", "[Database][Enumerator]") {
+    auto defaultColl = C4DatabaseTest::getCollection(db, kC4DefaultCollectionSpec);
+    auto populateDB  = [&](unsigned recordCount) -> void {
+        TransactionHelper t(db);
+
+        slice       body        = R"({"name":{"first":"Lue","last":"Laserna"}})"_sl;
+        alloc_slice encodedBody = c4db_encodeJSON(db, body, ERROR_INFO());
+        REQUIRE(encodedBody);
+        unsigned numDocs = 0;
+        for ( int i = 0; i < recordCount; ++i ) {
+            string docID = litecore::format("doc%07u", ++numDocs);
+            // Save document:
+            C4DocPutRequest rq      = {};
+            rq.docID                = slice(docID);
+            rq.allocedBody          = {(void*)encodedBody.buf, encodedBody.size};
+            rq.save                 = true;
+            c4::ref<C4Document> doc = c4coll_putDoc(defaultColl, &rq, nullptr, ERROR_INFO());
+            REQUIRE(doc != nullptr);
+        }
+    };
+    populateDB(10000);
+
+    C4EnumeratorOptions options = kC4DefaultEnumeratorOptions;
+    options.flags &= ~kC4IncludeBodies;
+    options.flags &= ~kC4IncludeNonConflicted;
+    options.flags |= kC4IncludeDeleted;
+
+    // With the above options, particularly the option to include only the conflicted documents,
+    // the following option, kC4Unsorted, becomes very significant in terms of performance. This
+    // is because the option to get sorted result will kepp SQLite from using, or taking advantage of,
+    // the index we have on DocumentFlags kConflicted. It is most salient when there are a large
+    // number of documents with only few having conflicts.
+    // Motivated by DBAccess::unresolvedDocsEnumerator, c.f. CBL-4506
+
+    auto doIt = [&](bool sorted) -> double {
+        if ( sorted ) {
+            options.flags &= ~kC4Unsorted;
+        } else {
+            options.flags |= kC4Unsorted;
+        }
+        c4::ref<C4DocEnumerator> e = c4coll_enumerateAllDocs(defaultColl, &options, ERROR_INFO());
+        REQUIRE(e);
+        unsigned  count = 0;
+        C4Error   error;
+        Stopwatch sw;
+        while ( c4enum_next(e, ERROR_INFO(error)) ) { ++count; }
+        double elapsed = sw.lap();
+        REQUIRE(error == C4Error{});
+        REQUIRE(count == 0);
+        return elapsed;
+    };
+    double elapsedUnsorted = doIt(false);
+    double elapsedSorted   = doIt(true);
+    cout << "Enum with conflicted/sorted takes " << elapsedSorted << " sec, Enum With conflicted/unsorted takes "
+         << elapsedUnsorted << "sec, ratio = " << elapsedSorted / elapsedUnsorted << endl;
+    auto config = c4db_getConfig2(db);
+    if ( config->encryptionKey.algorithm != kC4EncryptionNone ) {
+        // Encrypted case is not stable.
+        return;
+    }
+    CHECK(elapsedSorted / elapsedUnsorted > 5);
```

Comment:
Could this be flaky when running on some machines?

Reply:
I picked number, 5, quite conservatively. On my machine, the ratio is always > 10, but I found over the weekend that jenkins' windows build may be smaller. The theory is, the more docs the db has, the higher the ratio, I don't want to put too much time on this test. Perhaps this test does not belong here, but I wanted to set a guard here lest in the future we somehow we make changes to cause the problem reappear.

Reply:
It sounds like a performance test, perhaps it should be run as part of the perf suite for showfast?

Reply:
Definitely. I should get familiar with that.